app/controllers/registrations_controller.rb

```diff
@@ -545,6 +545,7 @@ def stripe_webhook
           #   and Stripe tries again after an exponential backoff. So we (erroneously!) record the creation timestamp
           #   in our DB _after_ the backed-off event has been processed. This can lead to a wrong registration order :(
           stored_payment.update!(created_at: audit_event.created_at_remote)
+          stored_intent.holder.add_history_entry({ payment_status: stored_intent.wca_status, iso_amount: ruby_money.cents }, "stripe_webhook", audit_event.id, 'Payment')
```

Comment:
Hmm... `Registration` already has a `record_payment` method, which currently is nothing more than a glorified "insert new registration_payment entity".

Could you look into possibly augmenting that method, so that it automatically also creates an history entry?
Or even more, could you consider adding `on_create` hooks to `RegistrationPayment` that automatically record some history entry? (The challenge there is probably going to be to pass along the webhook event that triggered the history write, so don't try too hard if the solution isn't obvious :sweat_smile: )

Reply:
makes sense, sounds easy to do